sycamore/transforms/map.py

```diff
@@ -8,7 +8,16 @@
 from sycamore.plan_nodes import Node, UnaryNode
 
 
+def rename(new_function_name: str):
+    def decorator(f):
+        f.__name__ = new_function_name
+        return f
+
+    return decorator
+
+
 def generate_map_function(f: Callable[[Document], Document]) -> Callable[[dict[str, Any]], dict[str, Any]]:
+    @rename(f.__name__)
```

Comment:
What is f.__name__ if you pass in an anonymous function or like a lambda?

Reply:
A lambda function will show up as `<lambda>`